### PR TITLE
Tuchfarber/enterprise email template 1

### PR DESCRIPTION
**Pull request by tuchfarber**
**Description:** 
First step in making the email notification template attached to the enterprise instead of the site. This makes the site field nullable, adds the enterprise_customer field, and changes all tests to reference the new field. I have verified that this is NOT being used in prod or stage currently.

**JIRA:** Link to JIRA ticket
[WL-1240]

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 
No dependencies. Will need two follow up PRs (removing field from model, running removal migration)

**Merge deadline:** List merge deadline (if any)
N/A

**Installation instructions:** List any non-trivial installation instructions.
N/A

**Testing instructions:**

(Available on the WL sandbox right now)
1. Add enrollment notification email template in django admin
2. See that it attaches to enterprise customer instead of site.

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

File: enterprise/migrations/0027_auto_20170921_1201.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            name='enterprise_customer',
+            field=models.OneToOneField(related_name='enterprise_enrollment_template', default=None, to='enterprise.EnterpriseCustomer'),
+            preserve_default=False,
+        ),
+        migrations.AddField(
+            model_name='historicalenrollmentnotificationemailtemplate',
+            name='enterprise_customer',
+            field=models.ForeignKey(related_name='+', on_delete=django.db.models.deletion.DO_NOTHING, db_constraint=False, blank=True, to='enterprise.EnterpriseCustomer', null=True),
+        ),
+        migrations.AlterField(
```
Comment (UmanShahzad) [inline enterprise/migrations/0027_auto_20170921_1201.py]: @douglashall Do we have to do the whole safe-deployment shenanigans for altering fields too?

File: enterprise/models.py [hunk 2]
```diff
@@ -996,8 +1007,8 @@ def __str__(self):
         """
         Return human-readable string representation.
         """
-        return '<EnrollmentNotificationEmailTemplate for site with ID {}>'.format(
-            self.site.id
+        return '<EnrollmentNotificationEmailTemplate for EnterpriseCustomer with ID {}>'.format(
```
Comment (UmanShahzad) [inline enterprise/models.py]: Nit: `ID` -> `UUID`

**Comment by UmanShahzad** [comment]
> Will need two follow up PRs (removing field from model, running removal migration)

@tuchfarber Thanks a lot for making this safe move!

**Comment by UmanShahzad** [comment]
@tuchfarber This isn't squashed and doesn't contain a changelog / version bump -- was that an accident? Would you like to revert it or is this intentional?

**Comment by tuchfarber** [comment]
It was an accident. Haven't done a revert yet, is it an easy process?

**Comment by tuchfarber** [comment]
^ @UmanShahzad

**Comment by UmanShahzad** [comment]
@tuchfarber Yep, there's a revert button on the merge line above. I've made https://github.com/edx/edx-enterprise/pull/200

**Pull request merged**